### PR TITLE
[RESTEASY-2116] Versions upgrade - mail, vertx, undertow, rxjava, validator

### PR DESCRIPTION
**Pull request by rsvoboda**
Versions upgrade - mail, vertx, undertow, rxjava, validator

https://issues.jboss.org/browse/RESTEASY-2116

### CODE REVIEW

File: resteasy-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -22,15 +22,15 @@
...
         <version.org.glassfish.jaxb.jaxb>2.3.1</version.org.glassfish.jaxb.jaxb>
         <version.com.sun.xml.fastinfoset>1.2.13</version.com.sun.xml.fastinfoset>
         <version.commons-io.commons-io>2.5</version.commons-io.commons-io>
         <version.hamcrest>1.3</version.hamcrest>
         <version.io.netty.netty>3.10.6.Final</version.io.netty.netty>
         <version.io.netty.netty4>4.1.31.Final</version.io.netty.netty4>
-        <version.io.vertx>3.5.4</version.io.vertx>
-        <version.io.undertow>1.4.18.Final</version.io.undertow>
+        <version.io.vertx>3.6.2</version.io.vertx>
+        <version.io.undertow>1.4.26.Final</version.io.undertow>
```
Comment (jbliznak) [inline resteasy-dependencies-bom/pom.xml]: Why not move directly to undertow 2.x.x ? WildFly 12+ is using this stream.

**Comment by rsvoboda** [comment]
One failed combination due to [ERROR] Plugin org.apache.maven.plugins:maven-resources-plugin:2.7 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-resources-plugin:jar:2.7: Could not transfer artifact org.apache.maven.plugins:maven-resources-plugin:pom:2.7 from/to central (https://repo.maven.apache.org/maven2): Connection reset -> [Help 1]

Seems like TravisCI env issue

File: resteasy-dependencies-bom/pom.xml (hunk 1 above)
Comment (rsvoboda) [reply resteasy-dependencies-bom/pom.xml]: RESTEasy project has Undertow integration so it's not just about WF based version of Undertow - https://github.com/resteasy/Resteasy/tree/master/server-adapters/resteasy-undertow. I didn't want to do major bumps in this PR.

@asoldano Can we jump to 2.0.z series ?

Comment (jbliznak) [reply resteasy-dependencies-bom/pom.xml]: Sure but there is only one integration class in resteasy and undertow 2.0.z should be compatible with 1.4.z https://abi-laboratory.pro/index.php?view=archives_report&lang=java&l=undertow-core&v1=1.4.26&v2=2.0.16

Comment (rsvoboda) [reply resteasy-dependencies-bom/pom.xml]: Moving to Undertow 2

Comment (asoldano) [reply resteasy-dependencies-bom/pom.xml]: fine, thanks

**Comment by asoldano** [comment]
> One failed combination due to [ERROR] Plugin org.apache.maven.plugins:maven-resources-plugin:2.7 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-resources-plugin:jar:2.7: Could not transfer artifact org.apache.maven.plugins:maven-resources-plugin:pom:2.7 from/to central (https://repo.maven.apache.org/maven2): Connection reset -> [Help 1]
> 
> Seems like TravisCI env issue

it is, right

**Comment by asoldano** [comment]
merged, thanks

**Pull request merged**